rotkehlchen/chain/substrate/manager.py

```diff
@@ -0,0 +1,169 @@
+import logging
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    DefaultDict,
+    Dict,
+    Iterable,
+    List,
+    NamedTuple,
+    Optional,
+    Set,
+    Tuple,
+    Union,
+    cast,
+    overload,
+)
+
+from scalecodec.base import RuntimeConfiguration
+from scalecodec.type_registry import load_type_registry_preset
+from substrateinterface import Keypair, SubstrateInterface
+from substrateinterface.exceptions import SubstrateRequestException
+from typing_extensions import Literal
+
+from rotkehlchen.constants.assets import A_KSM
+from rotkehlchen.constants.misc import ZERO
+from rotkehlchen.fval import FVal
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.typing import KusamaAddress
+from rotkehlchen.user_messages import MessagesAggregator
+
+from .typing import KusamaNodeName, SubstrateChain, SubstrateChainData, SubstrateNode
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+"""
+TODO priority 1:
+- Check whether we need to set up the RuntimeConfiguration and if so, how to
+set up multiple RuntimeConfiguration. Otherwise check the statement that keeps
+the decodes up to date:
+  * https://github.com/polkascan/py-substrate-interface#keeping-type-registry-presets-up-to-date
+  * Fix `ExtrinsicMetadata` error
+  * https://github.com/polkascan/py-scale-codec/blob/3db9ac54ae467c0c653a85c9224bce7ba38f778f/scalecodec/base.py#L381
+  * https://github.com/polkascan/py-scale-codec#multiple-runtime-configurations
+- Manually test cases:
+  - Add/edit/delete KSM account/accounts
+  - Validate balances and totals (use accounts without KSM as well)
+- Improve node connection (check how EthereumManager does):
```

Comment:
I'll ask you more about this tomorrow. At the moment I have the parity node hard coded, but we have 2 nodes available: Parity and Web3 Foundation

Reply:
Done, pending to add "own node" in the mix